runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/commits/StreamingEngineWorkCommitter.java

```diff
@@ -214,11 +216,13 @@ private boolean tryAddToCommitBatch(Commit commit, CommitWorkStream.RequestBatch
     return isCommitAccepted;
   }
 
-  // Helper to batch additional commits into the commit batch as long as they fit.
-  // Returns a commit that was removed from the queue but not consumed or null.
-  private Commit expandBatch(CommitWorkStream.RequestBatcher batcher) {
+  /**
+   * Helper to batch additional commits into the commit batch as long as they fit. Returns a commit
+   * that was removed from the queue but not consumed or null.
+   */
+  private @Nullable Commit expandBatch(CommitWorkStream.RequestBatcher batcher) {
     int commits = 1;
-    while (true) {
+    while (isRunning.get()) {
```

Comment:
nit: I'd just remove this one, it adds overhead to streams with many keys and we're already checking between each batch

Reply:
done